### DISCUSSION TITLE
V coding style coming from JS/Deno

### DISCUSSION THREAD

**Discussion by Brixy** [vlang/v#21034] [Questions and Answers] [answered]
Hello guys,

As I have mentioned elsewhere, I have rewritten all my scripts from JS/Deno to V—and it was a phantastic experience.

Now I have a question about the coding style. The script in question should be split into small functions, yet it is not big enough to split it into modules and files. It reads encrypted files into an array and writes them into a CSV file:

```
struct App {
mut:
	// define variables
}

fn read_and_decrypt_files(mut app App) ! {
	// …
}

fn read_lines(mut app App) ! {
	// read line by line; used later
}

fn read_secrets_into_csv(mut app App) ! {
	// read secrets
	read_lines(mut app)! // nested function; lines will become CSV columns
 }

fn write_csv_file(mut app App) ! {
	// …
}

fn main() {
	mut app := App{}
	read_and_decrypt_files(mut app)!
	read_secrets_into_csv(mut app)!
	write_csv_file(mut app)!
}
```

Is this the preferred V coding style for using variables across functions, or would you suggest using global variables, `pub …` or anything else.

Thanks a lot!

**Comment by JalonSolov** [comment, marked as answer]
V doesn't recommend globals at all.  While you can have them, they require both an extra keyword _and_ a command line switch to enable them.

What you have will work fine, but V supports the idea of [methods](https://github.com/vlang/v/blob/master/doc/docs.md#methods) which can be used to modify struct values, so you don't have to pass the struct around to all the functions.

**Reply by Brixy** [reply]
Ahh! Got it. Thanks a lot for pointing out the methods section.
